### PR TITLE
refactor: remove SchemaCache from ApiRequest

### PR DESCRIPTION
**Pull request by steve-chavez**
Refactor needed for https://github.com/PostgREST/postgrest/pull/1582. Solves what's commented on https://github.com/PostgREST/postgrest/issues/1804#issuecomment-1267591660.

### CODE REVIEW

File: src/PostgREST/ApiRequest.hs [hunk 1]
```diff
@@ -270,98 +265,52 @@ getRanges method QueryParams{qsOrder,qsRanges} hdrs
...
-    ActionMutate MutationUpdate -> qsColumns
-    ActionInvoke InvPost        -> qsColumns
-    _                           -> Nothing
-
-  payloadColumns =
-    case (contentMediaType, action) of
-      (_, ActionInvoke InvGet)  -> S.fromList $ fst <$> qsParams
-      (_, ActionInvoke InvHead) -> S.fromList $ fst <$> qsParams
-      (MTUrlEncoded, _)         -> S.fromList $ map (T.decodeUtf8 . fst) $ parseSimpleQuery $ LBS.toStrict reqBody
-      _ -> case (relevantPayload, columns) of
```
Comment (steve-chavez) [inline src/PostgREST/ApiRequest.hs:315]: See here

File: src/PostgREST/ApiRequest.hs [hunk 2]
```diff
@@ -270,98 +265,52 @@ getRanges method QueryParams{qsOrder,qsRanges} hdrs
...
-        then Right $ RawJSON reqBody
-        else note "All object keys must match" . payloadAttributes reqBody
-               =<< if LBS.null reqBody && pathIsProc
-                     then Right emptyObject
-                     else first BS.pack $ JSON.eitherDecode reqBody
-    (MTTextCSV, _) -> do
-      json <- csvToJson <$> first BS.pack (CSV.decodeByName reqBody)
-      note "All lines must have same number of fields" $ payloadAttributes (JSON.encode json) json
-    (MTUrlEncoded, _) ->
-      let paramsMap = HM.fromList $ (T.decodeUtf8 *** JSON.String . T.decodeUtf8) <$> parseSimpleQuery (LBS.toStrict reqBody) in
```
Comment (steve-chavez) [inline src/PostgREST/ApiRequest.hs:332]: ditto

File: src/PostgREST/ApiRequest.hs [hunk 3]
```diff
@@ -270,98 +265,52 @@ getRanges method QueryParams{qsOrder,qsRanges} hdrs
...
-    (ActionInvoke InvPost, _)              -> True
-    (ActionMutate MutationSingleUpsert, _) -> True
-    (ActionMutate MutationUpdate, _)       -> True
-    _                                      -> False
-  relevantPayload = case (contentMediaType, action) of
-    -- Though ActionInvoke GET/HEAD doesn't really have a payload, we use the payload variable as a way
-    -- to store the query string arguments to the function.
-    (_, ActionInvoke InvGet)             -> targetToJsonRpcParams (rightToMaybe target) qsParams
-    (_, ActionInvoke InvHead)            -> targetToJsonRpcParams (rightToMaybe target) qsParams
-    (MTUrlEncoded, ActionInvoke InvPost) -> targetToJsonRpcParams (rightToMaybe target) $ (T.decodeUtf8 *** T.decodeUtf8) <$> parseSimpleQuery (LBS.toStrict reqBody)
```
Comment (steve-chavez) [inline src/PostgREST/ApiRequest.hs:360]: ditto

**Review by steve-chavez — COMMENTED** [review]
I'm surprised that there's no LOC reduction. But there was noticeable code DRYed up. For example there were scattered `toStrict` done on Apirequest. This is now done on a single place.

Another benefit of this refactor is that the ApiRequest module is now streamlined.

File: src/PostgREST/ApiRequest.hs [hunk 4]
```diff
@@ -114,41 +112,10 @@ data PathInfo
...
-  toJSON (Fixed    v) = JSON.toJSON v
-  toJSON (Variadic v) = JSON.toJSON v
-
-toRpcParamValue :: ProcDescription -> (Text, Text) -> (Text, RpcParamValue)
-toRpcParamValue proc (k, v) | prmIsVariadic k = (k, Variadic [v])
-                            | otherwise       = (k, Fixed v)
-  where
-    prmIsVariadic prm = isJust $ find (\ProcParam{ppName, ppVar} -> ppName == prm && ppVar) $ pdParams proc
-
 -- | Convert rpc params `/rpc/func?a=val1&b=val2` to json `{"a": "val1", "b": "val2"}
```
Comment (laurenceisla) [inline src/PostgREST/ApiRequest.hs:118]: This comment should go before the function in the `CallPlan.hs` file, right?

**Review by laurenceisla — APPROVED** [review]

**Pull request merged**